lib/b64.c

```diff
@@ -231,10 +236,33 @@ jose_b64_dec_buf(const void *i, size_t il, void *o, size_t ol)
 
     for (size_t io = 0; io < il; io++) {
         uint8_t v = 0;
-
-        for (const char c = e[io]; v < len && c != map[v]; v++)
+        bool skip = false;
+        if (!seen_b64urlsafe)
```

Comment:
Why treat this in a special way ?
I would just set it in a case statement like for seen_b64.

I would also rename those var to:
seen_usrlsafe_char and seen_non_urlsafe_char ... we know it is *always* b64 so seen_b64 is kinda ambiguous